File: repository/templates/app-java/src/main/java/_Repository.java
package <%= appPackage %>.domain.repository.<%= repositoryPackageName %>;

import javax.inject.Inject;

public class <%= repositoryName %>Repository {

    <% if (remoteLocal.indexOf('local') >= 0) { %>@Inject
    <%= repositoryName %>LocalRepository localRepository;<% } %>

    <% if (remoteLocal.indexOf('remote') >= 0) { %>@Inject
    <%= repositoryName %>RemoteRepository remoteRepository;<% } %>

    @Inject
    public <%= repositoryName %>Repository() { }

  }

File: repository/templates/app-kotlin/src/main/java/_Repository.kt
package <%= appPackage %>.domain.repository.<%= repositoryPackageName %>;

import javax.inject.Inject;

class <%= repositoryName %>Repository @Inject constructor() {

    <% if (remoteLocal.indexOf('local') >= 0) { %>@Inject
    lateinit var localRepository: <%= repositoryName %>LocalRepository <% } %>

    <% if (remoteLocal.indexOf('remote') >= 0) { %>@Inject
    lateinit var remoteRepository: <%= repositoryName %>RemoteRepository<% } %>

  }
